Mathlib/GroupTheory/Perm/Fin.lean

```diff
@@ -279,6 +278,101 @@ end Fin
 
 end CycleRange
 
+section cycleIcc
+
+/-! ### `cycleIcc` section
+
+* Define the permutation `cycleIcc i j hij`, which is the cycle `(i i+1 .... j)` leaving
+`(0 ... i-1)` and `(j+1 ... n-1)` unchanged. In other words, it rotates elements in `[i, j]` one
+step to the right.
+-/
+
+namespace Fin
+
+local instance {n : ℕ} {i: Fin n} : NeZero (n - i.1) := NeZero.of_pos (by omega)
+
+variable {n : ℕ} {i j k : Fin n}
+
+/-- `cycleIcc i j hij` is the cycle `(i i+1 .... j)` leaving `(0 ... i-1)` and `(j+1 ... n-1)`
+unchanged. In other words, it rotates elements in `[i, j]` one step to the right.
+-/
+def cycleIcc (hij : i ≤ j): Perm (Fin n) :=
+  (cycleRange ((j - i).castLT (sub_val_lt_sub hij))).extendDomain
+  (natAdd_castLEEmb n (sub_le_right i)).toEquivRange
+
+theorem cycleIcc_lt (hij : i ≤ j) (h : k < i) : (cycleIcc hij) k = k :=
+  Perm.extendDomain_apply_not_subtype ((j - i).castLT _).cycleRange
+    (natAdd_castLEEmb n _).toEquivRange (by simp; omega)
+
+private lemma cycleIcc_aux (hij : i ≤ j) (h : i ≤ k) (kin : k ∈ Set.range ⇑(natAdd_castLEEmb n
+    (sub_le_right i))) : (cycleIcc hij) k = (natAdd_castLEEmb n (sub_le_right i))
+    (((j - i).castLT (sub_val_lt_sub hij)).cycleRange ((natAdd_castLEEmb n
+    (sub_le_right i)).toEquivRange.symm ⟨k, kin⟩)) := by
+  have kin : k ∈ Set.range ⇑(natAdd_castLEEmb n (sub_le_right i)) := by simp; omega
+  simp [cycleIcc, ((j - i).castLT (sub_val_lt_sub hij)).cycleRange.extendDomain_apply_subtype
+    (natAdd_castLEEmb n _).toEquivRange kin]
+
+private lemma cycleIcc_simp_lemma (h : i ≤ k) (kin : k ∈ Set.range ⇑(natAdd_castLEEmb n
+    (sub_le_right i))) : (((addNatEmb (n - (n - i.1))).trans
+    (finCongr _).toEmbedding).toEquivRange.symm ⟨k, kin⟩) = subNat i.1 (k.cast (by omega))
+    (by simp [h]) := by
+  simpa [symm_apply_eq] using eq_of_val_eq (by simp; omega)
+
+@[simp]
```

Comment:
Do you have an example where this `simp` actually has turned out to be useful? And hy the `.. _of_lt` does not serve a `simp`?